src/treemodel/position.js

```diff
@@ -406,7 +448,17 @@ CKEDITOR.define( [ 'treemodel/rootelement', 'utils', 'ckeditorerror' ], ( RootEl
 
 			path.push( offset );
 
-			return new Position( parent.root, path );
+			return new this( parent.root, path );
+		}
+
+		/**
+		 * Creates and returns a new instance of Position, which is equal to passed position.
+		 *
+		 * @param {treeModel.Position} position Position to be cloned.
+		 * @returns {treeModel.Position}
+		 */
+		static createFromPosition( position ) {
```

Comment:
This was explained during F2F discussion. More on this subject is here: https://github.com/ckeditor/ckeditor5-core/issues/64#issuecomment-163213464 (though I have to admit it may be difficult to understand). Long story short: `.this.clone()` is not going to work and we either introduce a parameter to `clone` method or use static creators. I think this solution is much cleaner and explicit regarding what happens behind scenes.

BTW. I was for `Position.clone` and `Range.clone` names so if you prefer those over `createFromPosition` and `createFromRange` we can discuss it. @pjasiun was against them.